### ISSUE TITLE
[Camera2D]  A Camera2D follower that follows around an IMovable

### ISSUE THREAD

**Issue by Taikatou** [MonoGame-Extended/Monogame-Extended#326]
I have implemented a Camera2D follower that follows around an IMovable. This follower allows users to put min and max barriers. (Values that the camera should not be able to see past) What do people think of this feature? I can see it being very useful. :D

**Comment by craftworkgames** [comment]
Yeah, we've had something like this on the to-do list for a while.

Submit a PR or paste some code here and we can discuss it more.

**Comment by AristurtleDev** [comment]
Thanks for the feature request. However, after reviewing I've decided not to implemented this. 

The reasoning for not implementing this is that the existing API already provides what users would need.  If a user needs a camera that follows anything, not just limited to `IMovable`, they can do this with just a few lines using the `Camera.LookAt()` method

```cs
// Add in update frmae
camera.LookAt(player.Position);
```

You can also use existing MonoGame utilities for smooth following with dampening

```cs
camera.Position = Vector2.Lerp(camera.Position, player.position, 0.1f);
```

Different games need different behaviors. Camera following varies wildly across game types and genres.  Any system we create would either be too simple for some use cases or too complex for others.  By keeping the implementation on the game developer side, they maintain full control over exactly how it works.

Closing this as won't implement.  The existing `LookAt` method provides the ncessary foundation
